src/main/java/com/shopify/minishopify/model/Category.java

```diff
@@ -0,0 +1,50 @@
+package com.shopify.minishopify.model;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+import java.util.Objects;
+
+@Entity
+@Table(name = "category")
+public class Category {
+    @Id
+    @GeneratedValue
+    @Column(name = "id")
+    private int id;
+
+    @Column(name = "name")
+    private String name;
+
+    public Category() {
+        this.name = "";
+    }
+
+    public Category(String name) {
+        this.name = name;
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Category category = (Category) o;
+        return id == category.id && Objects.equals(name, category.name);
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
Why is this (and other overridden hashcode functions) necessary?

Reply:
convention is to not override equals unless hashcode is overriden as well. https://www.baeldung.com/java-equals-hashcode-contracts

Reply:
I think this is a postgres related thing. I remember seeing this somewhere.